server/main.lua

```diff
@@ -168,12 +168,27 @@ end)
 
 lib.callback.register('qbx_houserobbery:callback:checkPickup', function(source, houseIndex, pickupIndex)
     local playerCoords = GetEntityCoords(GetPlayerPed(source))
-    local pickup = sharedConfig.houses[houseIndex].pickups[pickupIndex]
-
-    if #(playerCoords - pickup.coords) > 3 then return end
-    if pickup.isBusy then exports.qbx_core:Notify(source, Lang:t('notify.busy')) return end
-    if pickup.isOpened then return end
-
+    if not sharedConfig or not sharedConfig.houses then
+        return
+    end
+    local house = sharedConfig.houses[houseIndex]
+    if not house or not house.pickups then
+        return
+    end
+    local pickup = house.pickups[pickupIndex]
+    if not pickup then
+        return
+    end
+    if #(playerCoords - pickup.coords) > 3 then
+        return
+    end
+    if pickup.isBusy then
+        exports.qbx_core:Notify(source, locale('notify.busy'))
+        return
+    end
+    if pickup.isOpened then
+        return
+    end
```

Comment:
```suggestion
    local house = sharedConfig.houses[houseIndex]
    if not house or not house.pickups then return end
    
    local pickup = house.pickups[pickupIndex]
    if not pickup or #(playerCoords - pickup.coords) > 3 then return end
    if pickup.isBusy then
        exports.qbx_core:Notify(source, locale('notify.busy'))
        return
    end
    if pickup.isOpened then return end
```

We shouldn't need to check for sharedConfig or the houses as if those are missing we are going to error well before this check. The other changes are mostly for readability and reducing the number of if statements we check against before returning